modules/latex.php

```diff
@@ -19,88 +19,464 @@
+			case -4 :
+				return '7pt'; // 5 in LaTeX
+			default :
+				return false;
+		}
+	}
+
+	/**
+	 * Workarounds for shortcomings in MathJax's rendering.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @param  strign $latex LaTeX expression.
+	 * @param  string  $fg   Foreground color.
+	 * @param  string  $bg   Background color.
+	 * @param  integer $s    Font size.
+	 * @return mixed         String if successful, otherwise false.
+	 */
+	public function mathjax_text_mode_workarounds( $latex, $fg, $bg, $s ) {
+		switch ( $latex ) {
+			case '\LaTeX' :
+			case '\TeX' :
+				return '$latex \mathrm{' . $latex . '}$';
+			case '\AmS' :
+			case '\AmS-\TeX' :
+			case '\AmS-\LaTeX' :
+				return $this->render_img( $latex, $fg, $bg, $s );
+			default:
+				return false;
+		}
+	}
+
+	/**
+	 * Print a valid MathJax expression.
+	 *
+	 * Wraps LaTeX expression with LaTeX font size command sequence when necessary.
+	 * Wraps expression in a span to control colors when necessary.
+	 *
+	 * @since 1.1
+	 *
+	 * @param  string  $latex LaTeX expression.
+	 * @param  string  $fg    Foreground color.
+	 * @param  string  $bg    Background color.
+	 * @param  integer $s     Font size. Default '0'.
+	 * @return string         LaTeX expression.
+	 */
+	public function render( $latex, $fg, $bg, $s = 0 ) {
+		// Output MathJax if it's not already on the page.
+		$this->output_mathjax_script();
+
+		$latex_size_pt   = $this->latex_size_pt( $s );
+		$styles          = array();
+		$text_mode_value = $this->mathjax_text_mode_workarounds( trim( $latex ), $fg, $bg, $s );
+
+		if ( $fg ) {
+			$styles[] = 'color: #' . $fg . ';';
+		}
+		if ( $bg ) {
+			$styles[] = 'background-color: #' . $bg . ';';
+		}
+		if ( $latex_size_pt ) {
+			$styles[] = 'font-size: ' . $latex_size_pt . ';';
+		}
 
-add_filter( 'no_texturize_shortcodes', 'latex_no_texturize' );
+		if ( $text_mode_value ) {
+			$output   = $text_mode_value;
+			$styles[] = 'line-height: 0;';
+		} else {
+			$output = sprintf('$latex %s$', $latex);
+		}
 
-add_filter( 'the_content', 'latex_markup', 9 ); // before wptexturize
-add_filter( 'comment_text', 'latex_markup', 9 ); // before wptexturize
-add_shortcode( 'latex', 'latex_shortcode' );
+		if ( $styles ) {
+			$output = sprintf('<span style="display: inline-block;%s">%s</span>',
+				implode('', $styles),
+				$output
+			);
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Render a LaTeX expression using WP Latex.
+	 *
+	 * To use for edge cases where MathJax can't render something.
+	 *
+	 * @since 3.8.0
+	 *
+	 * @param  string  $latex LaTeX expression.
+	 * @param  string  $fg    Foreground color.
+	 * @param  string  $bg    Background color.
+	 * @param  integer $s     Font size.
+	 * @return string
+	 */
+	public function render_img( $latex, $fg, $bg, $s = 0 ) {
+		$url = "//s0.wp.com/latex.php?latex=" . urlencode( $latex ) . "&bg=" . $bg . "&fg=" . $fg . "&s=" . $s;
+		$url = esc_url( $url );
+		list($width, $height) = getimagesize('http:' . $url);
```

Comment:
I like the idea of trying to figure out the size of the `<img />` to prevent layout shifts, but there are several problems with the implementation.
* `getimagesize()` accepts a raw URL, not an HTML-encoded one.
* `getimagesize()` can return `false`.
* `getimagesize()` won't be able to access remote URLs on some hosts.
* Most importantly: the use of `getimagesize()` on the server for every image for every page load won't scale.